api/model/src/main/java/org/projectnessie/model/NessieConstants.java

```diff
@@ -0,0 +1,23 @@
+/*
+ * Copyright (C) 2024 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.model;
+
+/** Utility class holding some constants, mostly to prevent potential class loader deadlocks. */
+public final class NessieConstants {
+  public static final Namespace EMPTY_NAMESPACE = Namespace.builder().build();
```

Comment:
This could probably be in a `static` inner class in `Namespace`... or do you prefer a separate class?

Reply:
That would not solve the issue, right?

Reply:
But it would. A `static` inner class is mostly equivalent to a top-level class (no extra loading dependencies), except it's namespaced under the outer class.

Reply:
That's not much different then. I thought you wanted to propose class-loading-lazy initialization from a static method in Namespace.